src/main/java/uk/gov/hmcts/dts/fact/entity/CourtDxCode.java

```diff
@@ -14,18 +14,23 @@
 @Setter
 @NoArgsConstructor
 public class CourtDxCode {
-    @Id()
+    @Id
     @SequenceGenerator(name = "seq-gen", sequenceName = "search_courtdxcode_id_seq", allocationSize = 1)
     @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seq-gen")
     private Integer id;
     @ManyToOne
     @JoinColumn(name = "court_id")
     private Court court;
-    @ManyToOne(cascade = CascadeType.ALL)
+    @OneToOne(cascade = CascadeType.ALL, orphanRemoval = true)
```

Comment:
Would this be a one to one? I would assume one court could have many dx codes? Unless we are storing it at one string comma delimited? :) 

Ignore me if I am being silly here prob missing something!

Reply:
the search_courtdxcode table can hold multiple rows for a court but the relationship between search_courtdxcode and search_dxcode is one to one